src/components/organisms/EventModal/EventModal.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useMemo } from "react";
+import Modal from "react-bootstrap/Modal";
+import { VenueEvent } from "types/venues";
+import { Room } from "types/rooms";
+
+import { WithVenueId } from "utils/id";
+import { enterVenue } from "utils/url";
+import { hasEventFinished, isEventLive } from "utils/event";
+import { formatUtcSecondsRelativeToNow } from "utils/time";
+
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+import { useRoom } from "hooks/useRoom";
+
+import { Button } from "components/atoms/Button";
+
+import "./EventModal.scss";
+
+const emptyRoom: Room = {
+  title: "",
+  subtitle: "",
+  url: "",
+  about: "",
+  x_percent: 0,
+  y_percent: 0,
+  width_percent: 0,
+  height_percent: 0,
+  isEnabled: false,
+  image_url: "",
+};
+
+export interface EventModalProps {
+  show: boolean;
+  onHide: () => void;
+  event: WithVenueId<VenueEvent>;
+}
+
+export const EventModal: React.FC<EventModalProps> = ({
+  event,
+  onHide,
+  show,
+}) => {
+  const { currentVenue: eventVenue } = useRelatedVenues({
+    currentVenueId: event.venueId,
+  });
+
+  const eventRoom = useMemo(
+    () =>
+      eventVenue?.rooms?.find((room) => room.title === event.room) ?? emptyRoom,
+    [eventVenue, event]
+  );
+
+  const { enterRoom } = useRoom({
+    room: eventRoom,
+    venueName: eventVenue?.name ?? "",
+  });
+
+  const eventLocationToDisplay = event.room
+    ? event.room
+    : eventVenue?.name ?? event.venueId;
+
+  const goToEventLocation = () => {
+    onHide();
+
+    if (event.room) {
+      enterRoom();
+    } else {
+      enterVenue(event.venueId);
+    }
+  };
+
+  const isLive = isEventLive(event);
+
+  const getEventStatus = () => {
+    if (isLive) return `Happening now`;
+
+    if (hasEventFinished(event)) {
+      return `Ended`;
+    } else {
+      return `Starts ${formatUtcSecondsRelativeToNow(event.start_utc_seconds)}`;
+    }
+  };
+
+  return (
+    <Modal show={show} onHide={onHide} className="EventModal">
+      <div className="EventModal__content">
+        <h4 className="EventModal__title">{event.name}</h4>
+        <span className="EventModal__subtitle">
+          by {event.host} in{" "}
+          <button className="button--a" onClick={goToEventLocation}>
+            {eventLocationToDisplay}
+          </button>
```

Comment:
I think there is one more comment related to that. Let me explain everything in one place.
https://github.com/sparkletown/sparkle/pull/1426#discussion_r637920344

`<button />` here is a replacement for `<a />` tag. It behaves and looks like `a`, but the difference is that `a` requires `href` attribute, but here we have `onClick` handler only. For some reason, `LinkButton` didn't work for me as well. The location in the address bar was changing but the content did not rerender (I stayed in the same venue until I manually refresh the page).
So I decided to make a button with `onClick` handler that has same set of styles as `a` has, adding `button--a` class name.

\* `.button--a` is the underlined "Campfire" from the screenshot *
![image](https://user-images.githubusercontent.com/63313732/119663010-fc2c3d80-bdff-11eb-8045-1076e4e56b92.png)

Reply:
[NB] IMHO this should be done as a separate atomic component

Reply:
@0xdevalias :

> Hrmmm... not sure if this is something we want to introduce into global styles. I know there is a LinkButton component in one of @oleksii-front's PRs that I reviewed recently, plus I believe there is a 'text mode' on the existing Button component too (at least, it's props imply that there is) that would probably solve this use case better/properly, and not need us to add to global.scss (which is something we want to remove things from/eventually eliminate, rather than adding more to it)